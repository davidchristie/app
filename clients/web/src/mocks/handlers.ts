import { RequestHandler, rest } from "msw";

export const handlers: RequestHandler[] = [
  rest.get("/api/v1/auth/session", (request, response, context) => {
    return response(
      context.status(200),
      context.body(
        JSON.stringify({
          user: {
            id: "97406d59-7a49-4f1e-bb79-aba34cfcb405",
            name: "Test User",
            email: "test_user@email.com",
          },
        })
      )
    );
  }),
];
