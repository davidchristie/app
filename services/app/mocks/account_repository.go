// Code generated by MockGen. DO NOT EDIT.
// Source: github.com/davidchristie/app/services/app/repositories (interfaces: AccountRepository)

// Package mocks is a generated GoMock package.
package mocks

import (
	context "context"
	reflect "reflect"

	entities "github.com/davidchristie/app/services/app/entities"
	gomock "github.com/golang/mock/gomock"
	uuid "github.com/google/uuid"
)

// MockAccountRepository is a mock of AccountRepository interface.
type MockAccountRepository struct {
	ctrl     *gomock.Controller
	recorder *MockAccountRepositoryMockRecorder
}

// MockAccountRepositoryMockRecorder is the mock recorder for MockAccountRepository.
type MockAccountRepositoryMockRecorder struct {
	mock *MockAccountRepository
}

// NewMockAccountRepository creates a new mock instance.
func NewMockAccountRepository(ctrl *gomock.Controller) *MockAccountRepository {
	mock := &MockAccountRepository{ctrl: ctrl}
	mock.recorder = &MockAccountRepositoryMockRecorder{mock}
	return mock
}

// EXPECT returns an object that allows the caller to indicate expected use.
func (m *MockAccountRepository) EXPECT() *MockAccountRepositoryMockRecorder {
	return m.recorder
}

// FindByID mocks base method.
func (m *MockAccountRepository) FindByID(arg0 context.Context, arg1 uuid.UUID) (*entities.Account, error) {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "FindByID", arg0, arg1)
	ret0, _ := ret[0].(*entities.Account)
	ret1, _ := ret[1].(error)
	return ret0, ret1
}

// FindByID indicates an expected call of FindByID.
func (mr *MockAccountRepositoryMockRecorder) FindByID(arg0, arg1 interface{}) *gomock.Call {
	mr.mock.ctrl.T.Helper()
	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "FindByID", reflect.TypeOf((*MockAccountRepository)(nil).FindByID), arg0, arg1)
}

// FindByProvider mocks base method.
func (m *MockAccountRepository) FindByProvider(arg0 context.Context, arg1, arg2, arg3 string) (*entities.Account, error) {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "FindByProvider", arg0, arg1, arg2, arg3)
	ret0, _ := ret[0].(*entities.Account)
	ret1, _ := ret[1].(error)
	return ret0, ret1
}

// FindByProvider indicates an expected call of FindByProvider.
func (mr *MockAccountRepositoryMockRecorder) FindByProvider(arg0, arg1, arg2, arg3 interface{}) *gomock.Call {
	mr.mock.ctrl.T.Helper()
	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "FindByProvider", reflect.TypeOf((*MockAccountRepository)(nil).FindByProvider), arg0, arg1, arg2, arg3)
}

// Insert mocks base method.
func (m *MockAccountRepository) Insert(arg0 context.Context, arg1 *entities.Account) error {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "Insert", arg0, arg1)
	ret0, _ := ret[0].(error)
	return ret0
}

// Insert indicates an expected call of Insert.
func (mr *MockAccountRepositoryMockRecorder) Insert(arg0, arg1 interface{}) *gomock.Call {
	mr.mock.ctrl.T.Helper()
	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "Insert", reflect.TypeOf((*MockAccountRepository)(nil).Insert), arg0, arg1)
}
