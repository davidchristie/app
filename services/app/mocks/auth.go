// Code generated by MockGen. DO NOT EDIT.
// Source: github.com/davidchristie/app/services/app/auth (interfaces: Auth)

// Package mocks is a generated GoMock package.
package mocks

import (
	context "context"
	reflect "reflect"

	auth "github.com/davidchristie/app/services/app/auth"
	gomock "github.com/golang/mock/gomock"
)

// MockAuth is a mock of Auth interface.
type MockAuth struct {
	ctrl     *gomock.Controller
	recorder *MockAuthMockRecorder
}

// MockAuthMockRecorder is the mock recorder for MockAuth.
type MockAuthMockRecorder struct {
	mock *MockAuth
}

// NewMockAuth creates a new mock instance.
func NewMockAuth(ctrl *gomock.Controller) *MockAuth {
	mock := &MockAuth{ctrl: ctrl}
	mock.recorder = &MockAuthMockRecorder{mock}
	return mock
}

// EXPECT returns an object that allows the caller to indicate expected use.
func (m *MockAuth) EXPECT() *MockAuthMockRecorder {
	return m.recorder
}

// Authorize mocks base method.
func (m *MockAuth) Authorize(arg0 string) (*auth.AuthorizeResult, error) {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "Authorize", arg0)
	ret0, _ := ret[0].(*auth.AuthorizeResult)
	ret1, _ := ret[1].(error)
	return ret0, ret1
}

// Authorize indicates an expected call of Authorize.
func (mr *MockAuthMockRecorder) Authorize(arg0 interface{}) *gomock.Call {
	mr.mock.ctrl.T.Helper()
	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "Authorize", reflect.TypeOf((*MockAuth)(nil).Authorize), arg0)
}

// Callback mocks base method.
func (m *MockAuth) Callback(arg0 context.Context, arg1, arg2, arg3 string) (*auth.CallbackResult, error) {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "Callback", arg0, arg1, arg2, arg3)
	ret0, _ := ret[0].(*auth.CallbackResult)
	ret1, _ := ret[1].(error)
	return ret0, ret1
}

// Callback indicates an expected call of Callback.
func (mr *MockAuthMockRecorder) Callback(arg0, arg1, arg2, arg3 interface{}) *gomock.Call {
	mr.mock.ctrl.T.Helper()
	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "Callback", reflect.TypeOf((*MockAuth)(nil).Callback), arg0, arg1, arg2, arg3)
}

// Session mocks base method.
func (m *MockAuth) Session(arg0 context.Context, arg1 string) (*auth.Session, error) {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "Session", arg0, arg1)
	ret0, _ := ret[0].(*auth.Session)
	ret1, _ := ret[1].(error)
	return ret0, ret1
}

// Session indicates an expected call of Session.
func (mr *MockAuthMockRecorder) Session(arg0, arg1 interface{}) *gomock.Call {
	mr.mock.ctrl.T.Helper()
	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "Session", reflect.TypeOf((*MockAuth)(nil).Session), arg0, arg1)
}
