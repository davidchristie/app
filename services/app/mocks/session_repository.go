// Code generated by MockGen. DO NOT EDIT.
// Source: github.com/davidchristie/app/services/app/repositories (interfaces: SessionRepository)

// Package mocks is a generated GoMock package.
package mocks

import (
	context "context"
	reflect "reflect"

	entities "github.com/davidchristie/app/services/app/entities"
	gomock "github.com/golang/mock/gomock"
	uuid "github.com/google/uuid"
)

// MockSessionRepository is a mock of SessionRepository interface.
type MockSessionRepository struct {
	ctrl     *gomock.Controller
	recorder *MockSessionRepositoryMockRecorder
}

// MockSessionRepositoryMockRecorder is the mock recorder for MockSessionRepository.
type MockSessionRepositoryMockRecorder struct {
	mock *MockSessionRepository
}

// NewMockSessionRepository creates a new mock instance.
func NewMockSessionRepository(ctrl *gomock.Controller) *MockSessionRepository {
	mock := &MockSessionRepository{ctrl: ctrl}
	mock.recorder = &MockSessionRepositoryMockRecorder{mock}
	return mock
}

// EXPECT returns an object that allows the caller to indicate expected use.
func (m *MockSessionRepository) EXPECT() *MockSessionRepositoryMockRecorder {
	return m.recorder
}

// FindByID mocks base method.
func (m *MockSessionRepository) FindByID(arg0 context.Context, arg1 uuid.UUID) (*entities.Session, error) {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "FindByID", arg0, arg1)
	ret0, _ := ret[0].(*entities.Session)
	ret1, _ := ret[1].(error)
	return ret0, ret1
}

// FindByID indicates an expected call of FindByID.
func (mr *MockSessionRepositoryMockRecorder) FindByID(arg0, arg1 interface{}) *gomock.Call {
	mr.mock.ctrl.T.Helper()
	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "FindByID", reflect.TypeOf((*MockSessionRepository)(nil).FindByID), arg0, arg1)
}

// FindBySessionToken mocks base method.
func (m *MockSessionRepository) FindBySessionToken(arg0 context.Context, arg1 string) (*entities.Session, error) {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "FindBySessionToken", arg0, arg1)
	ret0, _ := ret[0].(*entities.Session)
	ret1, _ := ret[1].(error)
	return ret0, ret1
}

// FindBySessionToken indicates an expected call of FindBySessionToken.
func (mr *MockSessionRepositoryMockRecorder) FindBySessionToken(arg0, arg1 interface{}) *gomock.Call {
	mr.mock.ctrl.T.Helper()
	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "FindBySessionToken", reflect.TypeOf((*MockSessionRepository)(nil).FindBySessionToken), arg0, arg1)
}

// Insert mocks base method.
func (m *MockSessionRepository) Insert(arg0 context.Context, arg1 *entities.Session) error {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "Insert", arg0, arg1)
	ret0, _ := ret[0].(error)
	return ret0
}

// Insert indicates an expected call of Insert.
func (mr *MockSessionRepositoryMockRecorder) Insert(arg0, arg1 interface{}) *gomock.Call {
	mr.mock.ctrl.T.Helper()
	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "Insert", reflect.TypeOf((*MockSessionRepository)(nil).Insert), arg0, arg1)
}
