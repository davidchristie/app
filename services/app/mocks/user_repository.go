// Code generated by MockGen. DO NOT EDIT.
// Source: github.com/davidchristie/app/services/app/repositories (interfaces: UserRepository)

// Package mocks is a generated GoMock package.
package mocks

import (
	context "context"
	reflect "reflect"

	entities "github.com/davidchristie/app/services/app/entities"
	gomock "github.com/golang/mock/gomock"
	uuid "github.com/google/uuid"
)

// MockUserRepository is a mock of UserRepository interface.
type MockUserRepository struct {
	ctrl     *gomock.Controller
	recorder *MockUserRepositoryMockRecorder
}

// MockUserRepositoryMockRecorder is the mock recorder for MockUserRepository.
type MockUserRepositoryMockRecorder struct {
	mock *MockUserRepository
}

// NewMockUserRepository creates a new mock instance.
func NewMockUserRepository(ctrl *gomock.Controller) *MockUserRepository {
	mock := &MockUserRepository{ctrl: ctrl}
	mock.recorder = &MockUserRepositoryMockRecorder{mock}
	return mock
}

// EXPECT returns an object that allows the caller to indicate expected use.
func (m *MockUserRepository) EXPECT() *MockUserRepositoryMockRecorder {
	return m.recorder
}

// FindByID mocks base method.
func (m *MockUserRepository) FindByID(arg0 context.Context, arg1 uuid.UUID) (*entities.User, error) {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "FindByID", arg0, arg1)
	ret0, _ := ret[0].(*entities.User)
	ret1, _ := ret[1].(error)
	return ret0, ret1
}

// FindByID indicates an expected call of FindByID.
func (mr *MockUserRepositoryMockRecorder) FindByID(arg0, arg1 interface{}) *gomock.Call {
	mr.mock.ctrl.T.Helper()
	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "FindByID", reflect.TypeOf((*MockUserRepository)(nil).FindByID), arg0, arg1)
}

// FindByPrimaryEmail mocks base method.
func (m *MockUserRepository) FindByPrimaryEmail(arg0 context.Context, arg1 string) (*entities.User, error) {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "FindByPrimaryEmail", arg0, arg1)
	ret0, _ := ret[0].(*entities.User)
	ret1, _ := ret[1].(error)
	return ret0, ret1
}

// FindByPrimaryEmail indicates an expected call of FindByPrimaryEmail.
func (mr *MockUserRepositoryMockRecorder) FindByPrimaryEmail(arg0, arg1 interface{}) *gomock.Call {
	mr.mock.ctrl.T.Helper()
	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "FindByPrimaryEmail", reflect.TypeOf((*MockUserRepository)(nil).FindByPrimaryEmail), arg0, arg1)
}

// Insert mocks base method.
func (m *MockUserRepository) Insert(arg0 context.Context, arg1 *entities.User) error {
	m.ctrl.T.Helper()
	ret := m.ctrl.Call(m, "Insert", arg0, arg1)
	ret0, _ := ret[0].(error)
	return ret0
}

// Insert indicates an expected call of Insert.
func (mr *MockUserRepositoryMockRecorder) Insert(arg0, arg1 interface{}) *gomock.Call {
	mr.mock.ctrl.T.Helper()
	return mr.mock.ctrl.RecordCallWithMethodType(mr.mock, "Insert", reflect.TypeOf((*MockUserRepository)(nil).Insert), arg0, arg1)
}
