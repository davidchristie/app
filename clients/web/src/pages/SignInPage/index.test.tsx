import { render, screen } from "../../testing";
import { signedInSession } from "../../testing/data";
import { MockApiProvider } from "../../testing/mocks";
import { SignInPage } from ".";

describe("SignInPage", () => {
  it("renders correctly", () => {
    const { container } = render(
      <MockApiProvider data={{ session: signedInSession }}>
        <SignInPage />
      </MockApiProvider>
    );
    const element = screen.getByTestId("SignInPage");
    expect(element).toBe(container.firstChild);
    expect(element).toMatchSnapshot();
  });

  it("can sign in with GitHub", () => {
    const authorize = jest.fn();
    render(
      <MockApiProvider data={{ authorize, session: signedInSession }}>
        <SignInPage />
      </MockApiProvider>
    );
    screen.getByTestId("SignInPage__authorizeButton-github").click();
    expect(authorize).toBeCalledTimes(1);
    expect(authorize).toBeCalledWith("github");
  });
});
