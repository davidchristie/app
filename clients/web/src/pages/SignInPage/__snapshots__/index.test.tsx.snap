// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignInPage renders correctly 1`] = `
<div
  class="MuiBox-root MuiBox-root-1"
  data-testid="SignInPage"
>
  <div
    class="MuiBox-root MuiBox-root-2"
    data-testid="Page"
  >
    <header
      class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorDefault MuiPaper-elevation4"
      data-testid="AppBar"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
      >
        <div
          class="MuiBox-root MuiBox-root-5"
        >
          <a
            class="MuiTypography-root makeStyles-title-3 MuiTypography-h6"
            href="/"
          >
            App
          </a>
          <div
            class="MuiBox-root MuiBox-root-6"
          >
            <a
              aria-disabled="false"
              class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-colorInherit"
              data-testid="AppBar__signInButton"
              href="/signin"
              role="button"
              tabindex="0"
            >
              <span
                class="MuiButton-label"
              >
                Sign in
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
    </header>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg"
      data-testid="MainContent"
    >
      <div
        class="MuiBox-root MuiBox-root-7"
      >
        <h2
          class="MuiTypography-root MuiTypography-h2"
          data-testid="PageHeading"
        >
          Sign In
        </h2>
        <div
          class="MuiBox-root MuiBox-root-8"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained"
            data-testid="SignInPage__GitHubAuthorizeButton"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Sign in with GitHub
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
        <div
          class="MuiBox-root MuiBox-root-9"
        >
          <a
            aria-disabled="false"
            class="MuiButtonBase-root MuiButton-root MuiButton-contained"
            data-testid="SignInPage__GoogleAuthorizeButton"
            role="button"
            tabindex="0"
          >
            <span
              class="MuiButton-label"
            >
              Sign in with Google
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
